Feisty/GameDetails/GameDetailsViewController.swift

```diff
@@ -37,9 +40,64 @@ class GameDetailsViewController: UIViewController {
     viewModel.getGameData()
     setUpGameCardDropShadow()
     
+    setUpShoppingCartFloatingActionButton()
+    setUpAddToCartFloatingActionButton()
+    
+  }
+  
+  private func setUpShoppingCartFloatingActionButton() {
+    
     shoppingCartFAB.backgroundColor = UIColor.clear
-    addToCartFAB.backgroundColor = UIColor.clear
     shoppingCartFAB.button.setImage(#imageLiteral(resourceName: "ShoppingCart"), for: .normal)
+    shoppingCartFAB.actionToPerform = {
+      
+      let storyboard = UIStoryboard(name: "Main", bundle: .main)
+      let className = ShoppingCartViewController.className
+      let shoppingModal = storyboard.instantiateViewController(withIdentifier: className) as? ShoppingCartViewController
+      
+      guard let shoppingCartViewController = shoppingModal else {
+        return
+      }
+      
+      shoppingCartViewController.shoppingCart = self.shoppingCart
+      
+      self.present(shoppingCartViewController, animated: true, completion: nil)
+    }
+    
+  }
+  
+  private func setUpAddToCartFloatingActionButton() {
+    
+    addToCartFAB.backgroundColor = UIColor.clear
+    addToCartFAB.actionToPerform = { [weak self] in
+      
+      guard let self = self else {
+        return
+      }
+      
+      guard let game = self.selectedGame else {
+        return
+      }
+      
+      self.shoppingCart?.shoppingList.append(game)
+      
+      self.shoppingCartRightConstraint.constant = 32
+      
+      let animations: () -> Void = { [weak self] in
+        self?.view.layoutIfNeeded()
+      }
+      
+      let completion: (Bool) -> Void = { [weak self] completed in
```

Comment:
Bit confusing, why does this take a boolean but not actually use `completed` ?

Reply:
That's what the closure for animate requires. I did however use a "_" instead of completed.